src/MessageHandler/VirusScanHandler.php

```diff
@@ -0,0 +1,79 @@
+<?php
+
+namespace App\MessageHandler;
+
+use Doctrine\ORM\EntityManagerInterface;
+use Psr\Log\LoggerInterface;
+use Symfony\Component\Messenger\Handler\MessageHandlerInterface;
+
+use App\Message\VirusScan;
+use App\Entity\File;
+use App\Repository\FileRepository;
+
+class VirusScanHandler implements MessageHandlerInterface
+{
+    /**
+     * The Entity Manager.
+     *
+     * @var EntityManagerInterface
+     */
+    private $entityManager;
+
+    /**
+     * The File Repository.
+     *
+     * @var FileRepository
+     */
+    private $fileRepository;
+
+    /**
+     * The monolog logger.
+     *
+     * @var LoggerInterface
+     */
+    private $logger;
+
+    /**
+     * Constructor for this Controller, to set up default services.
+     *
+     * @param EntityManagerInterface   $entityManager           The entity handler.
+     * @param FileRepository           $fileRepository          The file Repository.
+     * @param LoggerInterface          $virusScanLogger         Name hinted virus_scan logger.
+     */
+    public function __construct(EntityManagerInterface $entityManager, FileRepository $fileRepository, LoggerInterface $virusScanLogger)
+    {
+        $this->entityManager = $entityManager;
+        $this->fileRepository = $fileRepository;
+        $this->logger = $virusScanLogger;
+    }
+
+    /**
+     * Invoke function to scan a file for viruses.
+     *
+     * @param VirusScan $virusScan The VirusScan message to be handled.
+     */
+    public function __invoke(VirusScan $virusScan)
+    {
+        $fileId = $virusScan->getFileId();
+        $file = $this->fileRepository->find($fileId);
+        if ($file instanceof File) {
+            $filePath = $file->getFilePath();
+            try {
+                $socket = (new \Socket\Raw\Factory())->createClient('unix:///run/clamd.scan/clamd.sock');
```

Comment:
This shouldn't be hard-coded as it can change per installation.

Reply:
You can put this in the .env file.